src/output/output_format.cpp

```diff
@@ -131,4 +133,19 @@ void Bin1_format::print_match(const Hsp_context& r, const Metadata &metadata, Te
 		out.write((uint32_t)r.subject_id);
 		out.write(r.bit_score() / std::max((unsigned)r.query.source().length(), r.subject_len));
 	}
-}
\ No newline at end of file
+}
+
+void Clustering_format::print_query_intro(size_t query_num, const char *query_name, unsigned query_len, TextBuffer &out, bool unaligned) const {
+	out.write((uint32_t)query_num);
+}
+
+void Clustering_format::print_match(const Hsp_context& r, const Metadata &metadata, TextBuffer &out) {
+	if (r.query_id <= r.subject_id) {
+		out.write((uint32_t)r.subject_id);
+		// TODO: enable other clustering measures
```

Comment:
Agreed, this should be configurable by the user. Other measures could be the bit score, the negative log of the evalue, or the bit score divided by the length of the longer sequence. It would also be interesting to have the Needleman-Wunsch global alignment score and sequence identity here, although Diamond can't compute this at the moment.

Reply:
would the Needleman-Wunsch ga-score also be stored in the Hsp_context structure (or something else that is accessible from the print_match function?

Reply:
all of them and more have now been implemented